cabal-install/Distribution/Client/ProjectBuilding.hs

```diff
@@ -977,7 +977,9 @@ buildAndInstallUnpackedPackage verbosity
             -- While this breaks the prefix-relocatable property of the lirbaries
             -- it is necessary on macOS to stay under the load command limit of the
             -- macOS mach-o linker. See also @PackageHash.hashedInstalledPackageIdVeryShort@.
-            otherFiles <- filter (not . isPrefixOf entryDir) <$> listFilesRecursive tmpDir
+            -- We also normalise paths to ensure that there are no different representations
+            -- for the same path. Like / and \\ on windows under msys.
+            otherFiles <- filter (not . isPrefixOf entryDir . normalise) <$> listFilesRecursive tmpDir
```

Comment:
isn't it sufficient to normalize just `tmpDir` here? So we only have to apply the function once instead of for every entry. The rest of the path is created by `</>` which will keep it consistent.